File: packages/expo-dev-launcher/ios/Modules/DevLauncherExceptions.swift
// Copyright 2015-present 650 Industries. All rights reserved.

import ExpoModulesCore

final class DevLauncherInvalidURLException: Exception, @unchecked Sendable {
  override var reason: String {
    "Invalid url provided to loadApp"
  }
}

final class DevLauncherLoadAppException: GenericException<String>, @unchecked Sendable {
  override var reason: String {
    "Failed to load app: \(param)"
  }
}

File: packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/modules/DevLauncherExceptions.kt
package expo.modules.devlauncher.modules

import expo.modules.kotlin.exception.CodedException

internal class DevLauncherInvalidURLException :
  CodedException("Invalid url provided to loadApp")

internal class DevLauncherNotAvailableException :
  CodedException("DevLauncherController is not available")

internal class DevLauncherLoadAppException(cause: Throwable?) :
  CodedException("Failed to load app", cause)
